packages/foundry/contracts/modules/RepayCloseModule.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity >=0.8.13;
+
+import "../LadleStorage.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
+import "@yield-protocol/utils-v2/contracts/token/TransferHelper.sol";
+
+contract RepayCloseModule is LadleStorage {
+    using CastU256I128 for uint256;
+    using CastU256U128 for uint256;
+    using TransferHelper for IERC20;
+
+    ///@notice Creates a Healer module for the Ladle
+    ///@param cauldron_ address of the Cauldron
+    ///@param weth_ address of WETH
+    constructor(ICauldron cauldron_, IWETH9 weth_) LadleStorage(cauldron_, weth_) {}
+
+    /// @dev Use fyToken in the Ladle to repay debt. Return unused fyToken to `to`.
+    /// Return as much collateral as debt was repaid, as well. This function is only used when
+    /// removing liquidity added with "Borrow and Pool", so it's safe to assume the exchange rate
+    /// is 1:1. If used in other contexts, it might revert, which is fine.
+    function repayFromLadle(bytes12 vaultId_, address collateralReceiver, address remainderReceiver)
+        external payable
+        returns (uint256 repaid)
+    {
+        (bytes12 vaultId, DataTypes.Vault memory vault) = getVault(vaultId_);
+        DataTypes.Series memory series = getSeries(vault.seriesId);
+        DataTypes.Balances memory balances = cauldron.balances(vaultId);
+        
+        uint256 amount = series.fyToken.balanceOf(address(this));
+        repaid = amount <= balances.art ? amount : balances.art;
+
+        // Update accounting, burn fyToken and return collateral
+        if (repaid > 0) {
+            cauldron.pour(vaultId, -(repaid.i128()), -(repaid.i128()));
+            series.fyToken.burn(address(this), repaid);
+            IJoin ilkJoin = getJoin(vault.ilkId);
+            ilkJoin.exit(collateralReceiver, repaid.u128());
+        }
+
+        // Return remainder
+        if (amount - repaid > 0) IERC20(address(series.fyToken)).safeTransfer(remainderReceiver, amount - repaid);
+
+    }
+
+    /// @dev Use base in the Ladle to repay debt. Return unused base to `to`.
+    /// Return as much collateral as debt was repaid, as well. This function is only used when
+    /// removing liquidity added with "Borrow and Pool", so it's safe to assume the exchange rate
+    /// is 1:1. If used in other contexts, it might revert, which is fine.
+    function closeFromLadle(bytes12 vaultId_, address collateralReceiver, address remainderReceiver)
+        external payable
+        returns (uint256 repaid)
+    {
+        (bytes12 vaultId, DataTypes.Vault memory vault) = getVault(vaultId_);
+        DataTypes.Series memory series = getSeries(vault.seriesId);
```

Comment:
You should actually check that `series.fyToken.underlying() == cauldron.assets(vault.ilkId)`, or some other cheaper method to check that this is only used for repaying "Borrow and Pool" vaults where the series underlying and the collateral are the same.